core/iwasm/compilation/aot_emit_aot_file.c

```diff
@@ -2047,6 +2047,11 @@ aot_resolve_target_info(AOTCompContext *comp_ctx, AOTObjectData *obj_data)
         }
 
         elf_header = (struct elf32_ehdr *)elf_buf;
+
+        /* Emit eXecute In Place file type while in indirect mode */
+        if (comp_ctx->is_indirect_mode)
+            elf_header->e_type = 4;
```

Comment:
Should also set e_type before line 2028 and before line 2071.
And had better "define E_TYPE_XIP 4", e.g. in line 1911, before `#define IMAGE_FILE_MACHINE_AMD64 0x8664`

Reply:
Done